subprojects/settings.gradle.kts

```diff
@@ -301,6 +301,61 @@ dependencyResolutionManagement {
     }
 }
 
+@Suppress("UnstableApiUsage")
+val avitoGithubRemoteCacheHost = settings.providers
+    .environmentVariable("GRADLE_CACHE_NODE_HOST")
+    .forUseAtConfigurationTime()
+
+val avitoGithubRemoteCachePush: String? by settings
+
+/**
+ * Included builds will inherit this cache config
+ * https://docs.gradle.org/current/userguide/build_cache.html#sec:build_cache_composite
+ *
+ * TODO it is not working as expected with --project-dir subprojects
+ *  problem is that's how IDE runs tasks from included builds by default
+ *  enabled only for "subprojects" for now
+ */
+buildCache {
+    remote<HttpBuildCache> {
+        url = remoteCacheUrl()
+        isEnabled = hasRemoteCacheUrl()
+        isPush = isRemoteCachePushEnabled()
+        isAllowUntrustedServer = true
+        isAllowInsecureProtocol = true
+    }
+}
+
+fun remoteCacheUrl(): java.net.URI {
+    val url = "http://${avitoGithubRemoteCacheHost.orNull}/cache/"
```

Comment:
I'd like to use more composable structure for properties, to make them more self-describable:

`avitoGithub.gradle.buildCache.remote.push`

Reply:
i agree, the problem is i wanted to use delegation property syntax `someVar by settings`, but it doesn't work with `.` in name
Not so much profit in this mechanism though, will change